encoding/thrift/outbound_nowire.go

```diff
@@ -0,0 +1,248 @@
+	if cc.ServiceName != "" {
+		svc = cc.ServiceName
+	}
+
+	if cc.Multiplexed {
+		p = multiplexedOutboundNoWireProtocol{
+			Protocol: p,
+			Service:  svc,
+		}
+	}
+
+	return noWireThriftClient{
+		p:             p,
+		cc:            c.ClientConfig,
+		thriftService: svc,
+		Enveloping:    cc.Enveloping,
+		NoWire:        cc.NoWire,
+	}
+}
+
+type noWireThriftClient struct {
+	cc transport.ClientConfig
+	p  stream.Protocol
+
+	// name of the Thrift service
+	thriftService string
+	Enveloping    bool
+	NoWire        bool
+}
+
+func (c noWireThriftClient) Call(ctx context.Context, reqBody stream.Enveloper, resBody stream.BodyReader, opts ...yarpc.CallOption) error {
+	// Code generated for Thrift client calls will probably be something like
+	// this:
+	//
+	// 	func (c *MyServiceClient) someMethod(ctx context.Context, arg1 Arg1Type, arg2 arg2Type, opts ...yarpc.CallOption) (returnValue, error) {
+	// 		var result myservice.SomeMethodResult
+	// 		args := myservice.SomeMethodHelper.Args(arg1, arg2)
+	// 		err := c.client.Call(ctx, args, result, opts...)
+	//
+	// 		success, err := myservice.SomeMethodHelper.UnwrapResponse(&result)
+	// 		return success, err
+	// 	}
+
+	out := c.cc.GetUnaryOutbound()
+
+	treq, proto, err := c.writeTransportRequest(reqBody)
+	if err != nil {
+		return err
+	}
+
+	call := encodingapi.NewOutboundCall(encoding.FromOptions(opts)...)
+	ctx, err = call.WriteToRequest(ctx, treq)
+	if err != nil {
+		return err
+	}
+
+	tres, err := out.Call(ctx, treq)
+	if err != nil {
+		return err
+	}
+	defer tres.Body.Close()
+
+	if _, err = call.ReadFromResponse(ctx, tres); err != nil {
+		return err
+	}
+
+	sr := proto.Reader(tres.Body)
+	defer sr.Close()
+
+	env, err := sr.ReadEnvelopeBegin()
+	if err != nil {
+		return errors.ResponseBodyDecodeError(treq, err)
+	}
+
+	switch env.Type {
+	case wire.Reply:
+		if err := resBody.Decode(sr); err != nil {
+			return err
+		}
+		return sr.ReadEnvelopeEnd()
+	case wire.Exception:
+		var exc internal.TApplicationException
+		if err = exc.Decode(sr); err != nil {
+			return errors.ResponseBodyDecodeError(treq, err)
+		}
+		return thriftException{
+			Service:   treq.Service,
+			Procedure: treq.Procedure,
+			Reason:    &exc,
+		}
+	default:
+		return errors.ResponseBodyDecodeError(
+			treq, errUnexpectedEnvelopeType(env.Type))
+	}
+}
+
+func (c noWireThriftClient) CallOneway(ctx context.Context, reqBody stream.Enveloper, opts ...yarpc.CallOption) (transport.Ack, error) {
+	out := c.cc.GetOnewayOutbound()
+
+	treq, _, err := c.writeTransportRequest(reqBody)
+	if err != nil {
+		return nil, err
+	}
+
+	call := encodingapi.NewOutboundCall(encoding.FromOptions(opts)...)
+	ctx, err = call.WriteToRequest(ctx, treq)
+	if err != nil {
+		return nil, err
+	}
+
+	return out.CallOneway(ctx, treq)
+}
+
+func (c noWireThriftClient) Enabled() bool {
+	return c.NoWire
+}
+
+func (c noWireThriftClient) writeTransportRequest(reqBody stream.Enveloper) (*transport.Request, stream.Protocol, error) {
```

Comment:
To be consistent with the outbound.go implementation, let's call this `buildTransportRequest`